modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/PythonClientCodegen.java

```diff
@@ -143,6 +159,46 @@ public void processOpts() {
     private static String dropDots(String str) {
         return str.replaceAll("\\.", "_");
     }
+    
+    @Override
+    public void postProcessParameter(CodegenParameter parameter){
+    	postProcessPattern(parameter.pattern, parameter.vendorExtensions);
+    }
+    
+    @Override
+    public void postProcessModelProperty(CodegenModel model, CodegenProperty property) {
+    	postProcessPattern(property.pattern, property.vendorExtensions);
+    }
+
+	/*
+	 * The swagger pattern spec follows the Perl convention and style of modifiers. Python
+	 * does not support this in as natural a way so it needs to convert it. See
+	 * https://docs.python.org/2/howto/regex.html#compilation-flags for details.
+	 */
+    public void postProcessPattern(String pattern, Map<String, Object> vendorExtensions){

```

Comment:
Create two vendor extensions from the pattern, one being the regex and the other a list of the modifiers.